metadata-ingestion/src/datahub/ingestion/source/unity/source.py

```diff
@@ -375,6 +382,18 @@ def get_workunits_internal(self) -> Iterable[MetadataWorkUnit]:
                 else:
                     raise ValueError("Unknown profiling config method")
 
+        # Collect cache statistics if tag processing was enabled
+        if self.config.include_tags and self.platform_resource_repository:
+            try:
+                self.report.tag_cache_info = get_unity_catalog_tag_cache_info(
+                    self.platform_resource_repository
+                )
+                logger.info(
+                    f"Unity Catalog tag cache statistics: {self.report.tag_cache_info}"
+                )
```

Comment:
we have some better patterns for this. I think it'd make sense to have the `PlatformResourceRepository` implement the `SupportsAsObj` protocol.

then, we can just include the platform_resource_repository directly in the report, and the `report.as_obj()` method will automatically call platform_resource_repository.as_obj(), which can then compute the stats on demand

the other alternative is to include the platform_resource_repository as a hidden field on the report and then override the compute_stats method to do this.

with the current approach, (1) the reporting code is spread all over the place and (2) the stats are only updated once